### ISSUE TITLE
Issue with data comming in 

### ISSUE THREAD

**Issue by killmasta93** [pfelk/pfelk#490]
Hi,
Currently recently installed PFELK, using the docker tutorial,
I currently check in the index management no data showing 
![image](https://github.com/pfelk/pfelk/assets/13953629/6a2ed71f-d9f3-4b06-8690-8d5cf928314a)

then i go to pfsense and checked the info this is what i have

![image](https://github.com/pfelk/pfelk/assets/13953629/d052f10b-68a1-475d-96ee-5055f34f87a1)
![image](https://github.com/pfelk/pfelk/assets/13953629/676f324a-78a6-498c-8df0-3f13b753396a)

when i go to dashboard i see this
![image](https://github.com/pfelk/pfelk/assets/13953629/118e095f-2d3e-4d33-9a38-5ca8530f7b81)

and this is what i have  in data views 
![image](https://github.com/pfelk/pfelk/assets/13953629/482aaff6-9ad0-462a-af82-de548f5ef63c)

Thank you

**Comment by a3ilson** [comment]
Are you using or referring to the pfelk/docker repo [here](https://github.com/pfelk/docker) or are you utilizing this repo and the docker-compose.yml (pfelk/pfelk)


It appears you were able to login and whatnot, did you update the 50-outputs.pfelk file with user and password? Assuming that is all setup and secure, does the password within 50-outputs.pfelk contain any special characters if so you'll need to escape any special characters.

**Comment by killmasta93** [comment]
Hi there, Thank you for the reply,  im using this repo 
` https://github.com/pfelk/docker/archive/refs/heads/main.zip`

the steps i took 

```
 sudo wget https://github.com/pfelk/docker/archive/refs/heads/main.zip
unzip main.zip
 sysctl -w vm.max_map_count=262144
cat .env

```
```

# User
ELASTIC_USER=elastic

# Password for the 'elastic' user (at least 6 characters)
ELASTIC_PASSWORD=thenewpass

# Password for the 'kibana_system' user (at least 6 characters)
KIBANA_PASSWORD=thenewpass

# Password for the 'logstash_system' user (at least 6 characters)
LOGSTASH_PASSWORD=thenewpass

# Version of Elastic products
STACK_VERSION=8.6.2

# Set the cluster name
CLUSTER_NAME=pfelk

# Set to 'basic' or 'trial' to automatically start the 30-day trial
LICENSE=basic
#LICENSE=trial

# Port to expose Elasticsearch HTTP API to the host
ES_PORT=9200
#ES_PORT=127.0.0.1:9200

# Port to expose Kibana to the host
KIBANA_PORT=5601
#KIBANA_PORT=80

# Increase or decrease based on the available host memory (in bytes)
MEM_LIMIT=1073741824

# Project namespace (defaults to the current folder name if not set)
#COMPOSE_PROJECT_NAME=myproject

```

```
  sed -i 's/logstash_system_password/LOGSTASH-PASSWORD/' etc/logstash/config/logstash.yml
 sed -i 's/elastic_password/ELASTIC-PASSWORD/' etc/pfelk/conf.d/50-outputs.pfelk
```
then docker compose up 

not sure if i missed something else?

Thank you again

**Comment by a3ilson** [comment]
Awe...it appears the pfelk/docker repo doesn't have the manual instructions for the roles. It appears you have everything configured but likely do not have the necessary permissions for Logstash to write to elastic. 

Reference the roles which may be created via CLI or manually via the web GUI. These are configured at step number 4, 4a and 5 within [here](https://github.com/pfelk/pfelk/blob/main/install/security.md)

**Comment by killmasta93** [comment]
Sorry for the late reply, didnt get the alert of the message, I was trying, and when i go in the docker container i try to run this command
so i ran the commands added the users 
but still showing the dashboard with the same issue would i need to go to step 7?
![image](https://github.com/pfelk/pfelk/assets/13953629/df9b69e0-580f-4a43-9e4e-7579e4f4ece2)


Thank you

**Comment by a3ilson** [comment]
Can you confirm that your 50 (outputs) file contains the credentials for elastic?

**Comment by a3ilson** [comment]
@killmasta93 
Update the pfelk.grok file and give it another try. I fixed a grok pattern issue where the pfSense would parse the OPNsense logs but if RFC5424 was utilized it failed to clean-up the dumb meta sequenceId values.

Merging this with issue #495

**Comment by killmasta93** [comment]
thanks for the reply, so after a while realizing the command 
```

sed -i 's/logstash_system_password/LOGSTASH-PASSWORD/' etc/logstash/config/logstash.yml
sed -i 's/elastic_password/ELASTIC-PASSWORD/' etc/pfelk/conf.d/50-outputs.pfelk
```

did not work, so i ran the 

`./set-logstash-password.sh`

which did update it, but i redid the whole steps again but now im getting 
`
docker-main-es01-1      | ERROR: [1] bootstrap checks failed. You must address the points described in the following [1] lines before starting Elasticsearch.`

**Comment by a3ilson** [comment]
thanks!

I'm actually in the process of testing and updating the pfelk/docker repo...

Check the /etc/logstash/config/logstash.yml and /etc/pfelk/conf.d/50-outputs.pfelk and see if they were updated with the password set in the .env file.

**Comment by killmasta93** [comment]
thanks yeah it got updated im attaching the info


```
root@dockers7:~/docker-main# cat .env 
# User
ELASTIC_USER=elastic

# Password for the 'elastic' user (at least 6 characters)
ELASTIC_PASSWORD=Passw0rd

# Password for the 'kibana_system' user (at least 6 characters)
KIBANA_PASSWORD=Passw0rd

# Password for the 'logstash_system' user (at least 6 characters)
LOGSTASH_PASSWORD=Passw0rd

# Version of Elastic products
STACK_VERSION=8.6.2

# Set the cluster name
CLUSTER_NAME=pfelk

# Set to 'basic' or 'trial' to automatically start the 30-day trial
LICENSE=basic
#LICENSE=trial

# Port to expose Elasticsearch HTTP API to the host
ES_PORT=9200
#ES_PORT=127.0.0.1:9200

# Port to expose Kibana to the host
KIBANA_PORT=5601
#KIBANA_PORT=80

# Increase or decrease based on the available host memory (in bytes)
MEM_LIMIT=1073741824

# Project namespace (defaults to the current folder name if not set)
#COMPOSE_PROJECT_NAME=myproject

```

```
root@dockers7:~/docker-main# cat etc/logstash/config/logstash.yml
# logstash.yml
################################################################################
# Version: 23.01                                                               #
# Required: Yes (DOCKER ONLY)                                                  #
# Description: This is a required file for a Docker installation               #
#                                                                              #
################################################################################
#
http.host: "0.0.0.0"
#path.config: etc/pfelk/logstash/pipeline/pipelines.yml
xpack.monitoring.elasticsearch.hosts: [ "https://es01:9200" ]
xpack.monitoring.elasticsearch.username: elastic
xpack.monitoring.elasticsearch.password: Passw0rd
xpack.monitoring.elasticsearch.ssl.certificate_authority: /usr/share/logstash/config/certs/ca/ca.crt

## X-Pack security credentials
xpack.monitoring.enabled: true

```


```
root@dockers7:~/docker-main# cat etc/pfelk/conf.d/50-outputs.pfelk
# 50-outputs.pfelk
################################################################################
# Version: 23.01(datastream)                                                   #
# Required: Yes                                                                #
# Description: Sends enriched logs to Elasticsearch.                           #
#                                                                              #
################################################################################
#
filter {
  if [log][syslog][appname] == "captiveportal" {
    mutate { add_field => { "[data_stream][namespace]" => "captiveportal" } }
  } else if [log][syslog][appname] == "dhcp" {
    mutate { add_field => { "[data_stream][namespace]" => "dhcp" } }
  } else if [log][syslog][appname] == "firewall" {
    mutate { add_field => { "[data_stream][namespace]" => "firewall" } }
  } else if "bind9" in [tags] or "dpinger" in [tags] or "ntpd" in [tags] or "web_portal" in [tags] {
    mutate { add_field => { "[data_stream][namespace]" => "firewall_processes" } }
  } else if [log][syslog][appname] == "haproxy" {
    mutate { add_field => { "[data_stream][namespace]" => "haproxy" } }
  } else if [log][syslog][appname] == "nginx" {
    mutate { add_field => { "[data_stream][namespace]" => "nginx" } }
  } else if [log][syslog][appname] == "openvpn" {
    mutate { add_field => { "[data_stream][namespace]" => "openvpn" } }
  } else if [log][syslog][appname] == "unbound" {
    mutate { add_field => { "[data_stream][namespace]" => "unbound" } }
  } else if [log][syslog][appname] == "suricata" {
    mutate { add_field => { "[data_stream][namespace]" => "suricata" } }
  } else if [log][syslog][appname] == "snort" {
    mutate { add_field => { "[data_stream][namespace]" => "snort" } }
  } else if [log][syslog][appname] == "squid" {
    mutate { add_field => { "[data_stream][namespace]" => "squid" } }
  } else {
    mutate { add_field => { "[data_stream][namespace]" => "unknown" } }
  }
}
output {
#  if [type] == "beats" {
#    elasticsearch {
#      hosts => ["http://es01:9200"]
#      index => "%{[@metadata][beat]}-%{[@metadata][version]}"
#      ecs_compatibility => "v1"
#      manage_template => false
#      ### X-Pack Username and Password ###
#      user => USERNAMEHERE
#      password => PASSWORDHERE
#    }
#  }
  elasticsearch {
    data_stream => "true"
    data_stream_type => "logs"
    data_stream_dataset => "pfelk"
    ###  Unsecured Method ###
    # hosts => ["http://es01:9200"]
    ### X-Pack Security Method ###
    hosts => ["https://es01:9200"]
    # ssl => true
    cacert => '/usr/share/logstash/config/certs/ca/ca.crt'
    user => "elastic"
    password => "Passw0rd"
  }
}
```
Thanks

**Comment by a3ilson** [comment]
@killmasta93 
I'll be archiving the pfelk/docker repo as I've finally merge it with the pfelk/pfelk repo. However, instructions are in-work...

1) I will have manual instructions posted today
2) Plan to incorporate into the pfelk-installer script as a option (i.e., install pfelk via docker)

The manual docker installation will be posted shortly and I'll add the link to this thread soon...I did not that you were looking at indices which should be empty. pfelk moved from indices to data streams a few iterations ago, please check the data streams tab
![Screen Shot 2023-08-13 at 4 04 59 PM](https://github.com/pfelk/pfelk/assets/16884679/05ce3abe-13c9-46a2-896d-c9e1433068f3)

**Comment by a3ilson** [comment]
@killmasta93 
Give this [guide](https://github.com/pfelk/pfelk/blob/main/install/docker.md) a try and let me know if it works out for you.

**Comment by killmasta93** [comment]
HI there thanks for the reply, so trying the guide found a typo 

`nano /etc/pfelk/doocker/.env`

which should be `nano /etc/pfelk/docker/.env`

i also deleted the images in docker just in case, 

so i reran the guide but after running the docker compose up -d 

i tried to access the web http://192.168.21.230:5601/ 

but im getting `{"statusCode":503,"error":"Service Unavailable","message":"License is not available."}`

i checked the .env 
```

root@dockers7:/etc/pfelk/docker# cat /etc/pfelk/docker/.env
# User
ELASTIC_USER=elastic

# Password for the 'elastic' user (at least 6 characters)
ELASTIC_PASSWORD=Passw0rd

# Password for the 'kibana_system' user (at least 6 characters)
KIBANA_PASSWORD=Passw0rd

# Password for the 'logstash_system' user (at least 6 characters)
LOGSTASH_PASSWORD=Passw0rd

# Version of Elastic products
STACK_VERSION=8.9.0

# Set the cluster name
CLUSTER_NAME=pfelk

# Set to 'basic' or 'trial' to automatically start the 30-day trial
LICENSE=basic
#LICENSE=trial

# Port to expose Elasticsearch HTTP API to the host
ES_PORT=9200
#ES_PORT=127.0.0.1:9200

# Port to expose Kibana to the host
KIBANA_PORT=5601
#KIBANA_PORT=80

# Increase or decrease based on the available host memory (in bytes)
MEM_LIMIT=1073741824

# Project namespace (defaults to the current folder name if not set)
#COMPOSE_PROJECT_NAME=myproject

```
i was checking the logs on the containers and found this


```
node.id=5lYtHlbDQ7iW5xfaihcSbA elasticsearch.node.name=es01 elasticsearch.cluster.name=pfelk

Authentication of [elastic] was terminated by realm [reserved] - failed to authenticate user [elastic] | @timestamp=2023-08-13T22:43:59.294Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es01][system_critical_read][T#1] log.logger=org.elasticsearch.xpack.security.authc.RealmsAuthenticator elasticsearch.cluster.uuid=TQGRpNlLQp-ghu87rZo_LQ elasticsearch.node.id=5lYtHlbDQ7iW5xfaihcSbA elasticsearch.node.name=es01 elasticsearch.cluster.name=pfelk

Authentication of [elastic] was terminated by realm [reserved] - failed to authenticate user [elastic] | @timestamp=2023-08-13T22:44:04.662Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es01][transport_worker][T#2] log.logger=org.elasticsearch.xpack.security.authc.RealmsAuthenticator elasticsearch.cluster.uuid=TQGRpNlLQp-ghu87rZo_LQ elasticsearch.node.id=5lYtHlbDQ7iW5xfaihcSbA elasticsearch.node.name=es01 elasticsearch.cluster.name=pfelk

Authentication of [elastic] was terminated by realm [reserved] - failed to authenticate user [elastic] | @timestamp=2023-08-13T22:44:27.461Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es01][generic][T#5] log.logger=org.elasticsearch.xpack.security.authc.RealmsAuthenticator elasticsearch.cluster.uuid=TQGRpNlLQp-ghu87rZo_LQ elasticsearch.node.id=5lYtHlbDQ7iW5xfaihcSbA elasticsearch.node.name=es01 elasticsearch.cluster.name=pfelk

Authentication of [elastic] was terminated by realm [reserved] - failed to authenticate user [elastic] | @timestamp=2023-08-13T22:44:27.460Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es01][system_critical_read][T#1] log.logger=org.elasticsearch.xpack.security.authc.RealmsAuthenticator elasticsearch.cluster.uuid=TQGRpNlLQp-ghu87rZo_LQ elasticsearch.node.id=5lYtHlbDQ7iW5xfaihcSbA elasticsearch.node.name=es01 elasticsearch.cluster.name=pfelk

Authentication of [elastic] was terminated by realm [reserved] - failed to authenticate user [elastic] | @timestamp=2023-08-13T22:44:57.397Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es01][transport_worker][T#1] log.logger=org.elasticsearch.xpack.security.authc.RealmsAuthenticator elasticsearch.cluster.uuid=TQGRpNlLQp-ghu87rZo_LQ elasticsearch.node.id=5lYtHlbDQ7iW5xfaihcSbA elasticsearch.node.name=es01 elasticsearch.cluster.name=pfelk

Authentication of [elastic] was terminated by realm [reserved] - failed to authenticate user [elastic] | @timestamp=2023-08-13T22:45:00.692Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es01][system_critical_read][T#1] log.logger=org.elasticsearch.xpack.security.authc.RealmsAuthenticator elasticsearch.cluster.uuid=TQGRpNlLQp-ghu87rZo_LQ elasticsearch.node.id=5lYtHlbDQ7iW5xfaihcSbA elasticsearch.node.name=es01 elasticsearch.cluster.name=pfelk

ERROR: Elasticsearch exited unexpectedly
```

i verified on the outputs the password is correct as in the .env
```

root@dockers7:/etc/pfelk/docker# cat /etc/pfelk/conf.d/50-outputs.pfelk
# 50-outputs.pfelk
################################################################################
# Version: 23.08a                                                              #
# Required: True                                                               #
# Description: Sends enriched logs to Elasticsearch.                           #
#                                                                              #
################################################################################
#
filter {
  if [log][syslog][appname] == "captiveportal" {
    mutate { add_field => { "[data_stream][namespace]" => "captiveportal" } }
  } else if [log][syslog][appname] == "dhcp" {
    mutate { add_field => { "[data_stream][namespace]" => "dhcp" } }
  } else if [log][syslog][appname] == "firewall" {
    mutate { add_field => { "[data_stream][namespace]" => "firewall" } }
  } else if "bind9" in [tags] or "dpinger" in [tags] or "ntpd" in [tags] or "web_portal" in [tags] {
    mutate { add_field => { "[data_stream][namespace]" => "firewall_processes" } }
  } else if [log][syslog][appname] == "haproxy" {
    mutate { add_field => { "[data_stream][namespace]" => "haproxy" } }
  } else if [log][syslog][appname] == "nginx" {
    mutate { add_field => { "[data_stream][namespace]" => "nginx" } }
  } else if [log][syslog][appname] == "openvpn" {
    mutate { add_field => { "[data_stream][namespace]" => "openvpn" } }
  } else if [log][syslog][appname] == "unbound" {
    mutate { add_field => { "[data_stream][namespace]" => "unbound" } }
  } else if [log][syslog][appname] == "suricata" {
    mutate { add_field => { "[data_stream][namespace]" => "suricata" } }
  } else if [log][syslog][appname] == "snort" {
    mutate { add_field => { "[data_stream][namespace]" => "snort" } }
  } else if [log][syslog][appname] == "squid" {
    mutate { add_field => { "[data_stream][namespace]" => "squid" } }
  } else {
    mutate { add_field => { "[data_stream][namespace]" => "unknown" } }
  }
}
output {
  elasticsearch {
    data_stream => "true"
    data_stream_type => "logs"
    data_stream_dataset => "pfelk"
    ###  Unsecured Method ###
    # hosts => ["http://localhost:9200"]
    ### X-Pack Security Method ###
    hosts => ["https://localhost:9200"]
    # ssl => true
    # [DOCKER] cacert => '/usr/share/logstash/config/certs/ca/ca.crt'
    # [DOCKER] user => "elastic"
    cacert => '/etc/logstash/config/certs/http_ca.crt' #[Disable if using Docker]
    user => "pfelk_logstash" #[Disable if using Docker]
    password => "Passw0rd"
  }
}
```

Thanks

**Comment by a3ilson** [comment]
I reinstalled and had no issue, but I didn't make any changes. I'm wondering if the password provided is having issues. 

I just found what is likely and hopefully the culprit/issue:
edit the 50-outputs.conf and revise as follows:
From:
```
    # [DOCKER] cacert => '/usr/share/logstash/config/certs/ca/ca.crt'
    # [DOCKER] user => "elastic"
    cacert => '/etc/logstash/config/certs/http_ca.crt' #[Disable if using Docker]
    user => "pfelk_logstash" #[Disable if using Docker]
    password => "Passw0rd"
```

To:
```
    cacert => '/usr/share/logstash/config/certs/ca/ca.crt'
    user => "elastic"
    # cacert => '/etc/logstash/config/certs/http_ca.crt' #[Disable if using Docker]
    # user => "pfelk_logstash" #[Disable if using Docker]
    password => "changeme"
```

**Comment by killmasta93** [comment]
Thanks for the reply, so quick question even if the the .env i have the password to Passw0rd I would leave unchanged on the 50-outputs.conf? would i also need to coment out this part?

```
    # cacert => '/etc/logstash/config/certs/http_ca.crt' #[Disable if using Docker]
    # user => "pfelk_logstash" #[Disable if using Docker]
```

Thank you

**Comment by a3ilson** [comment]
Update the 50-outputs.pfelk to the same password as that in the .env file.

**Comment by a3ilson** [comment]
So it should look like this:

```
    cacert => '/usr/share/logstash/config/certs/ca/ca.crt'
    user => "elastic"
    # cacert => '/etc/logstash/config/certs/http_ca.crt' #[Disable if using Docker]
    # user => "pfelk_logstash" #[Disable if using Docker]
    password => "passw0rd"
```

**Comment by killmasta93** [comment]
allright so got it working left for now the password the same as changeme

then i was putting up the templates then i see all of sudden the containers drop
the elastic nodes i checked the logs
the host itself has 20 gigs of ram with 8 CPU

![image](https://github.com/pfelk/pfelk/assets/13953629/2d427beb-50ea-4c21-8ec1-79de4a2e7c80)

![image](https://github.com/pfelk/pfelk/assets/13953629/dcff8102-1e7e-4766-9b45-cdba45a6004d)


```
attempts; for further information see https://www.elastic.co/guide/en/elasticsearch/reference/8.9/important-settings.html#initial_master_nodes | @timestamp=2023-08-14T01:19:55.394Z log.level=WARN ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=main log.logger=org.elasticsearch.cluster.coordination.ClusterBootstrapService elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

elected-as-master ([2] nodes joined in term 6)[_FINISH_ELECTION_, {es02}{Pm-jdMS-SEe9OWbFM7Gzxw}{bAZ3kIiPRK-LAPyPHjGVrw}{es02}{172.25.0.5}{172.25.0.5:9300}{cdfhilmrstw}{8.9.0} completing election, {es03}{qstbmZUwSBOKKqKMaUFylg}{N3NTb2bsRaaPgM22jfJB5A}{es03}{172.25.0.4}{172.25.0.4:9300}{cdfhilmrstw}{8.9.0} completing election], term: 6, version: 259, delta: master node changed {previous [], current [{es02}{Pm-jdMS-SEe9OWbFM7Gzxw}{bAZ3kIiPRK-LAPyPHjGVrw}{es02}{172.25.0.5}{172.25.0.5:9300}{cdfhilmrstw}{8.9.0}]}, added {{es03}{qstbmZUwSBOKKqKMaUFylg}{N3NTb2bsRaaPgM22jfJB5A}{es03}{172.25.0.4}{172.25.0.4:9300}{cdfhilmrstw}{8.9.0}} | @timestamp=2023-08-14T01:19:57.491Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es02][masterService#updateTask][T#1] log.logger=org.elasticsearch.cluster.service.MasterService elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

master node changed {previous [], current [{es02}{Pm-jdMS-SEe9OWbFM7Gzxw}{bAZ3kIiPRK-LAPyPHjGVrw}{es02}{172.25.0.5}{172.25.0.5:9300}{cdfhilmrstw}{8.9.0}]}, added {{es03}{qstbmZUwSBOKKqKMaUFylg}{N3NTb2bsRaaPgM22jfJB5A}{es03}{172.25.0.4}{172.25.0.4:9300}{cdfhilmrstw}{8.9.0}}, term: 6, version: 259, reason: Publication{term=6, version=259} | @timestamp=2023-08-14T01:19:58.439Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es02][clusterApplierService#updateTask][T#1] log.logger=org.elasticsearch.cluster.service.ClusterApplierService elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

starting file watcher ... | @timestamp=2023-08-14T01:19:58.626Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es02][clusterApplierService#updateTask][T#1] log.logger=org.elasticsearch.common.file.AbstractFileWatchingService elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

file settings service up and running [tid=69] | @timestamp=2023-08-14T01:19:58.712Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[file-watcher[/usr/share/elasticsearch/config/operator/settings.json]] log.logger=org.elasticsearch.common.file.AbstractFileWatchingService elasticsearch.cluster.uuid=OzrRC-XMSfu1myw8UIxGQA elasticsearch.node.id=Pm-jdMS-SEe9OWbFM7Gzxw elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

node-join: [{es02}{Pm-jdMS-SEe9OWbFM7Gzxw}{bAZ3kIiPRK-LAPyPHjGVrw}{es02}{172.25.0.5}{172.25.0.5:9300}{cdfhilmrstw}{8.9.0}] with reason [completing election] | @timestamp=2023-08-14T01:19:58.773Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es02][masterService#updateTask][T#1] log.logger=org.elasticsearch.cluster.coordination.NodeJoinExecutor elasticsearch.cluster.uuid=OzrRC-XMSfu1myw8UIxGQA elasticsearch.node.id=Pm-jdMS-SEe9OWbFM7Gzxw elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

publish_address {172.25.0.5:9200}, bound_addresses {0.0.0.0:9200} | @timestamp=2023-08-14T01:19:58.775Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=main log.logger=org.elasticsearch.http.AbstractHttpServerTransport elasticsearch.cluster.uuid=OzrRC-XMSfu1myw8UIxGQA elasticsearch.node.id=Pm-jdMS-SEe9OWbFM7Gzxw elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

node-join: [{es03}{qstbmZUwSBOKKqKMaUFylg}{N3NTb2bsRaaPgM22jfJB5A}{es03}{172.25.0.4}{172.25.0.4:9300}{cdfhilmrstw}{8.9.0}] with reason [completing election] | @timestamp=2023-08-14T01:19:58.777Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es02][masterService#updateTask][T#1] log.logger=org.elasticsearch.cluster.coordination.NodeJoinExecutor elasticsearch.cluster.uuid=OzrRC-XMSfu1myw8UIxGQA elasticsearch.node.id=Pm-jdMS-SEe9OWbFM7Gzxw elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

started {es02}{Pm-jdMS-SEe9OWbFM7Gzxw}{bAZ3kIiPRK-LAPyPHjGVrw}{es02}{172.25.0.5}{172.25.0.5:9300}{cdfhilmrstw}{8.9.0}{ml.machine_memory=1073741824, ml.allocated_processors=8, ml.max_jvm_size=536870912, ml.allocated_processors_double=8.0, xpack.installed=true} | @timestamp=2023-08-14T01:19:58.783Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=main log.logger=org.elasticsearch.node.Node elasticsearch.cluster.uuid=OzrRC-XMSfu1myw8UIxGQA elasticsearch.node.id=Pm-jdMS-SEe9OWbFM7Gzxw elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

license [5226af8f-bc60-491d-a12d-e4f58b9a718f] mode [basic] - valid | @timestamp=2023-08-14T01:20:02.387Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es02][clusterApplierService#updateTask][T#1] log.logger=org.elasticsearch.license.ClusterStateLicenseService elasticsearch.cluster.uuid=OzrRC-XMSfu1myw8UIxGQA elasticsearch.node.id=Pm-jdMS-SEe9OWbFM7Gzxw elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

license mode is [basic], currently licensed security realms are [reserved/reserved,file/default_file,native/default_native] | @timestamp=2023-08-14T01:20:02.393Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es02][clusterApplierService#updateTask][T#1] log.logger=org.elasticsearch.xpack.security.authc.Realms elasticsearch.cluster.uuid=OzrRC-XMSfu1myw8UIxGQA elasticsearch.node.id=Pm-jdMS-SEe9OWbFM7Gzxw elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

recovered [23] indices into cluster_state | @timestamp=2023-08-14T01:20:02.409Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es02][masterService#updateTask][T#1] log.logger=org.elasticsearch.gateway.GatewayService elasticsearch.cluster.uuid=OzrRC-XMSfu1myw8UIxGQA elasticsearch.node.id=Pm-jdMS-SEe9OWbFM7Gzxw elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

Node [{es02}{Pm-jdMS-SEe9OWbFM7Gzxw}] is selected as the current health node. | @timestamp=2023-08-14T01:20:08.646Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es02][management][T#2] log.logger=org.elasticsearch.health.node.selection.HealthNodeTaskExecutor elasticsearch.cluster.uuid=OzrRC-XMSfu1myw8UIxGQA elasticsearch.node.id=Pm-jdMS-SEe9OWbFM7Gzxw elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

Cluster health status changed from [RED] to [YELLOW] (reason: [shards started [[.internal.alerts-observability.apm.alerts-default-000001][0]]]). | @timestamp=2023-08-14T01:20:15.560Z log.level=INFO current.health=YELLOW previous.health=RED reason=shards started [[.internal.alerts-observability.apm.alerts-default-000001][0]] ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es02][masterService#updateTask][T#1] log.logger=org.elasticsearch.cluster.routing.allocation.AllocationService elasticsearch.cluster.uuid=OzrRC-XMSfu1myw8UIxGQA elasticsearch.node.id=Pm-jdMS-SEe9OWbFM7Gzxw elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

[gc][28] overhead, spent [332ms] collecting in the last [1.2s] | @timestamp=2023-08-14T01:20:18.242Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es02][scheduler][T#1] log.logger=org.elasticsearch.monitor.jvm.JvmGcMonitorService elasticsearch.cluster.uuid=OzrRC-XMSfu1myw8UIxGQA elasticsearch.node.id=Pm-jdMS-SEe9OWbFM7Gzxw elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

updating number_of_replicas to [2] for indices [.apm-source-map] | @timestamp=2023-08-14T01:20:21.347Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es02][masterService#updateTask][T#1] log.logger=org.elasticsearch.cluster.routing.allocation.AllocationService elasticsearch.cluster.uuid=OzrRC-XMSfu1myw8UIxGQA elasticsearch.node.id=Pm-jdMS-SEe9OWbFM7Gzxw elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

node-join[{es01}{8y3yCpxRS0ua_Ema9A-jlw}{LlsG3qu1RbSqeYllYfR79w}{es01}{172.25.0.6}{172.25.0.6:9300}{cdfhilmrstw}{8.9.0} joining], term: 6, version: 283, delta: added {{es01}{8y3yCpxRS0ua_Ema9A-jlw}{LlsG3qu1RbSqeYllYfR79w}{es01}{172.25.0.6}{172.25.0.6:9300}{cdfhilmrstw}{8.9.0}} | @timestamp=2023-08-14T01:20:21.353Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es02][masterService#updateTask][T#1] log.logger=org.elasticsearch.cluster.service.MasterService elasticsearch.cluster.uuid=OzrRC-XMSfu1myw8UIxGQA elasticsearch.node.id=Pm-jdMS-SEe9OWbFM7Gzxw elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

added {{es01}{8y3yCpxRS0ua_Ema9A-jlw}{LlsG3qu1RbSqeYllYfR79w}{es01}{172.25.0.6}{172.25.0.6:9300}{cdfhilmrstw}{8.9.0}}, term: 6, version: 283, reason: Publication{term=6, version=283} | @timestamp=2023-08-14T01:20:23.774Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es02][clusterApplierService#updateTask][T#1] log.logger=org.elasticsearch.cluster.service.ClusterApplierService elasticsearch.cluster.uuid=OzrRC-XMSfu1myw8UIxGQA elasticsearch.node.id=Pm-jdMS-SEe9OWbFM7Gzxw elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

node-join: [{es01}{8y3yCpxRS0ua_Ema9A-jlw}{LlsG3qu1RbSqeYllYfR79w}{es01}{172.25.0.6}{172.25.0.6:9300}{cdfhilmrstw}{8.9.0}] with reason [joining] | @timestamp=2023-08-14T01:20:23.826Z log.level=INFO ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es02][masterService#updateTask][T#1] log.logger=org.elasticsearch.cluster.coordination.NodeJoinExecutor elasticsearch.cluster.uuid=OzrRC-XMSfu1myw8UIxGQA elasticsearch.node.id=Pm-jdMS-SEe9OWbFM7Gzxw elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

[.kibana_security_session_1][0] marking unavailable shards as stale: [xD8K-L0YQDy8PD98Wg4I5A] | @timestamp=2023-08-14T01:20:24.060Z log.level=WARN ecs.version=1.2.0 service.name=ES_ECS event.dataset=elasticsearch.server process.thread.name=elasticsearch[es02][masterService#updateTask][T#1] log.logger=org.elasticsearch.cluster.routing.allocation.AllocationService elasticsearch.cluster.uuid=OzrRC-XMSfu1myw8UIxGQA elasticsearch.node.id=Pm-jdMS-SEe9OWbFM7Gzxw elasticsearch.node.name=es02 elasticsearch.cluster.name=pfelk

ERROR: Elasticsearch exited unexpectedly
```

**Comment by a3ilson** [comment]
I've updated the .env file...it was set for 1GB for each elasticsearch. I increased it to 4GB per node or 12GB. Please update the .env file and give it another try.

**Comment by a3ilson** [comment]
I also updated the docker.yml to define each containers memory (e.g. Kibana, logstash and elastic) which is defined in the .env (e.g., ES_MEM_LIMIT; KB_MEM_LIMIT; LS_MEM_LIMIT)

**Comment by killmasta93** [comment]
thanks there is a typo on the env had to change it to LS which is LG

got it work the web, but when i send the info from the pfsense to the PFELK i dont see no data, already ran these steps 

https://github.com/pfelk/pfelk/blob/main/install/templates.md#b-scripted-method-page_with_curl
![image](https://github.com/pfelk/pfelk/assets/13953629/ceee43d8-eac2-45c9-9710-cb7386457264)

i was checking the logs of logstash
```

[INFO ] 2023-08-14 02:37:37.869 [main] runner - JVM bootstrap flags: [-Dls.cgroup.cpuacct.path.override=/, -Dls.cgroup.cpu.path.override=/, -Xmx1G, -Xms1G, -Djruby.regexp.interruptible=true, -Djdk.io.File.enableADS=true, --add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED, --add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED, --add-exports=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED, --add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED, --add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED, --add-opens=java.base/java.security=ALL-UNNAMED, --add-opens=java.base/java.io=ALL-UNNAMED, --add-opens=java.base/java.nio.channels=ALL-UNNAMED, --add-opens=java.base/sun.nio.ch=ALL-UNNAMED, --add-opens=java.management/sun.management=ALL-UNNAMED]

/usr/share/logstash/vendor/bundle/jruby/2.6.0/gems/sinatra-2.2.4/lib/sinatra/base.rb:938: warning: constant Tilt::Cache is deprecated

[WARN ] 2023-08-14 02:37:40.425 [LogStash::Runner] pipelineregisterhook - Internal collectors option for Logstash monitoring is deprecated and targeted for removal in the next major version.

Please configure Metricbeat to monitor Logstash. Documentation can be found at: 

https://www.elastic.co/guide/en/logstash/current/monitoring-with-metricbeat.html

[INFO ] 2023-08-14 02:37:42.241 [LogStash::Runner] licensereader - Elasticsearch pool URLs updated {:changes=>{:removed=>[], :added=>[https://elastic:xxxxxx@es01:9200/]}}

[WARN ] 2023-08-14 02:37:42.821 [LogStash::Runner] licensereader - Restored connection to ES instance {:url=>"https://elastic:xxxxxx@es01:9200/"}

[INFO ] 2023-08-14 02:37:42.839 [LogStash::Runner] licensereader - Elasticsearch version determined (8.9.0) {:es_version=>8}

[WARN ] 2023-08-14 02:37:42.840 [LogStash::Runner] licensereader - Detected a 6.x and above cluster: the `type` event field won't be used to determine the document _type {:es_version=>8}

[INFO ] 2023-08-14 02:37:42.897 [LogStash::Runner] internalpipelinesource - Monitoring License OK

[INFO ] 2023-08-14 02:37:42.897 [LogStash::Runner] internalpipelinesource - Validated license for monitoring. Enabling monitoring pipeline.

[INFO ] 2023-08-14 02:37:43.209 [Api Webserver] agent - Successfully started Logstash API endpoint {:port=>9600, :ssl_enabled=>false}

[INFO ] 2023-08-14 02:37:44.619 [Converge PipelineAction::Create<.monitoring-logstash>] Reflections - Reflections took 428 ms to scan 1 urls, producing 132 keys and 464 values

[INFO ] 2023-08-14 02:37:45.051 [Converge PipelineAction::Create<.monitoring-logstash>] javapipeline - Pipeline `.monitoring-logstash` is configured with `pipeline.ecs_compatibility: v8` setting. All plugins in this pipeline will default to `ecs_compatibility => v8` unless explicitly configured otherwise.

[INFO ] 2023-08-14 02:37:45.089 [[.monitoring-logstash]-pipeline-manager] elasticsearchmonitoring - New Elasticsearch output {:class=>"LogStash::Outputs::ElasticSearchMonitoring", :hosts=>["https://es01:9200"]}

[INFO ] 2023-08-14 02:37:45.111 [[.monitoring-logstash]-pipeline-manager] elasticsearchmonitoring - Elasticsearch pool URLs updated {:changes=>{:removed=>[], :added=>[https://elastic:xxxxxx@es01:9200/]}}

[WARN ] 2023-08-14 02:37:45.255 [[.monitoring-logstash]-pipeline-manager] elasticsearchmonitoring - Restored connection to ES instance {:url=>"https://elastic:xxxxxx@es01:9200/"}

[INFO ] 2023-08-14 02:37:45.278 [[.monitoring-logstash]-pipeline-manager] elasticsearchmonitoring - Elasticsearch version determined (8.9.0) {:es_version=>8}

[WARN ] 2023-08-14 02:37:45.279 [[.monitoring-logstash]-pipeline-manager] elasticsearchmonitoring - Detected a 6.x and above cluster: the `type` event field won't be used to determine the document _type {:es_version=>8}

[WARN ] 2023-08-14 02:37:45.337 [[.monitoring-logstash]-pipeline-manager] javapipeline - 'pipeline.ordered' is enabled and is likely less efficient, consider disabling if preserving event order is not necessary

[INFO ] 2023-08-14 02:37:45.387 [[.monitoring-logstash]-pipeline-manager] javapipeline - Starting pipeline {:pipeline_id=>".monitoring-logstash", "pipeline.workers"=>1, "pipeline.batch.size"=>2, "pipeline.batch.delay"=>50, "pipeline.max_inflight"=>2, "pipeline.sources"=>["monitoring pipeline"], :thread=>"#<Thread:0x3fb46238@/usr/share/logstash/logstash-core/lib/logstash/java_pipeline.rb:134 run>"}

[INFO ] 2023-08-14 02:37:50.075 [[.monitoring-logstash]-pipeline-manager] javapipeline - Pipeline Java execution initialization time {"seconds"=>4.69}

[INFO ] 2023-08-14 02:37:50.102 [[.monitoring-logstash]-pipeline-manager] javapipeline - Pipeline started {"pipeline.id"=>".monitoring-logstash"}

[ERROR] 2023-08-14 02:38:11.477 [Converge PipelineAction::Create<pfelk>] elasticsearch - Invalid setting for elasticsearch output plugin:

  output {

    elasticsearch {

      # This setting must be a path

      # File does not exist or cannot be opened /etc/logstash/config/certs/http_ca.crt

      cacert => "/etc/logstash/config/certs/http_ca.crt"


```

**Comment by a3ilson** [comment]
Looks like the 50-outputs.pfelk changed on your end.

make sure you are using the two variables for docker. 

cacert => '/usr/share/logstash/config/certs/ca/ca.crt'
 user => "elastic"

**Comment by killmasta93** [comment]
thanks for the reply, ah yes my bad, i forgot to uncoment now i have this

```
root@dockers7:/etc/pfelk/conf.d# cat 50-outputs.pfelk 
# 50-outputs.pfelk
################################################################################
# Version: 23.08a                                                              #
# Required: True                                                               #
# Description: Sends enriched logs to Elasticsearch.                           #
#                                                                              #
################################################################################
#
filter {
  if [log][syslog][appname] == "captiveportal" {
    mutate { add_field => { "[data_stream][namespace]" => "captiveportal" } }
  } else if [log][syslog][appname] == "dhcp" {
    mutate { add_field => { "[data_stream][namespace]" => "dhcp" } }
  } else if [log][syslog][appname] == "firewall" {
    mutate { add_field => { "[data_stream][namespace]" => "firewall" } }
  } else if "bind9" in [tags] or "dpinger" in [tags] or "ntpd" in [tags] or "web_portal" in [tags] {
    mutate { add_field => { "[data_stream][namespace]" => "firewall_processes" } }
  } else if [log][syslog][appname] == "haproxy" {
    mutate { add_field => { "[data_stream][namespace]" => "haproxy" } }
  } else if [log][syslog][appname] == "nginx" {
    mutate { add_field => { "[data_stream][namespace]" => "nginx" } }
  } else if [log][syslog][appname] == "openvpn" {
    mutate { add_field => { "[data_stream][namespace]" => "openvpn" } }
  } else if [log][syslog][appname] == "unbound" {
    mutate { add_field => { "[data_stream][namespace]" => "unbound" } }
  } else if [log][syslog][appname] == "suricata" {
    mutate { add_field => { "[data_stream][namespace]" => "suricata" } }
  } else if [log][syslog][appname] == "snort" {
    mutate { add_field => { "[data_stream][namespace]" => "snort" } }
  } else if [log][syslog][appname] == "squid" {
    mutate { add_field => { "[data_stream][namespace]" => "squid" } }
  } else {
    mutate { add_field => { "[data_stream][namespace]" => "unknown" } }
  }
}
output {
  elasticsearch {
    data_stream => "true"
    data_stream_type => "logs"
    data_stream_dataset => "pfelk"
    ###  Unsecured Method ###
    # hosts => ["http://localhost:9200"]
    ### X-Pack Security Method ###
    hosts => ["https://localhost:9200"]
    # ssl => true
    cacert => '/usr/share/logstash/config/certs/ca/ca.crt'
    user => "elastic"
    #cacert => '/etc/logstash/config/certs/http_ca.crt' #[Disable if using Docker]
    #user => "pfelk_logstash" #[Disable if using Docker]
    password => "changeme"
  }
}
```

i checked the logs in the logstash, it seems that kibana is up running because i can access the dashboard

![image](https://github.com/pfelk/pfelk/assets/13953629/130699f5-95bf-4c73-a839-fa1ea4d9c01d)


```
essage=>"Elasticsearch Unreachable: [https://localhost:9200/][Manticore::SocketException] Connect to localhost:9200 [localhost/127.0.0.1] failed: Connection refused"}

[INFO ] 2023-08-14 16:16:27.362 [Ruby-0-Thread-19: /usr/share/logstash/vendor/bundle/jruby/2.6.0/gems/logstash-output-elasticsearch-11.15.9-java/lib/logstash/outputs/elasticsearch/http_client/pool.rb:213] elasticsearch - Failed to perform request {:message=>"Connect to localhost:9200 [localhost/127.0.0.1] failed: Connection refused", :exception=>Manticore::SocketException, :cause=>#<Java::OrgApacheHttpConn::HttpHostConnectException: Connect to localhost:9200 [localhost/127.0.0.1] failed: Connection refused>}

[WARN ] 2023-08-14 16:16:27.371 [Ruby-0-Thread-19: /usr/share/logstash/vendor/bundle/jruby/2.6.0/gems/logstash-output-elasticsearch-11.15.9-java/lib/logstash/outputs/elasticsearch/http_client/pool.rb:213] elasticsearch - Attempted to resurrect connection to dead ES instance, but got an error {:url=>"https://elastic:xxxxxx@localhost:9200/", :exception=>LogStash::Outputs::ElasticSearch::HttpClient::Pool::HostUnreachableError, :message=>"Elasticsearch Unreachable: [https://localhost:9200/][Manticore::SocketException] Connect to localhost:9200 [localhost/127.0.0.1] failed: Connection refused"}

[INFO ] 2023-08-14 16:16:32.399 [Ruby-0-Thread-19: /usr/share/logstash/vendor/bundle/jruby/2.6.0/gems/logstash-output-elasticsearch-11.15.9-java/lib/logstash/outputs/elasticsearch/http_client/pool.rb:213] elasticsearch - Failed to perform request {:message=>"Connect to localhost:9200 [localhost/127.0.0.1] failed: Connection refused", :exception=>Manticore::SocketException, :cause=>#<Java::OrgApacheHttpConn::HttpHostConnectException: Connect to localhost:9200 [localhost/127.0.0.1] failed: Connection refused>}

[WARN ] 2023-08-14 16:16:32.406 [Ruby-0-Thread-19: /usr/share/logstash/vendor/bundle/jruby/2.6.0/gems/logstash-output-elasticsearch-11.15.9-java/lib/logstash/outputs/elasticsearch/http_client/pool.rb:213] elasticsearch - Attempted to resurrect connection to dead ES instance, but got an error {:url=>"https://elastic:xxxxxx@localhost:9200/", :exception=>LogStash::Outputs::ElasticSearch::HttpClient::Pool::HostUnreachableError, :message=>"Elasticsearch Unreachable: [https://localhost:9200/][Manticore::SocketException] Connect to localhost:9200 [localhost/127.0.0.1] failed: Connection refused"}
```

after checking it seems that in the 50-outputs i changed localhost  to docker-es01-1 but now im getting another issue
```

[INFO ] 2023-08-14 16:31:01.603 [Ruby-0-Thread-46: /usr/share/logstash/vendor/bundle/jruby/2.6.0/gems/logstash-input-syslog-3.6.0/lib/logstash/inputs/syslog.rb:155] syslog - Starting syslog tcp listener {:address=>"0.0.0.0:5140"}

[WARN ] 2023-08-14 16:31:04.716 [Ruby-0-Thread-19: /usr/share/logstash/vendor/bundle/jruby/2.6.0/gems/logstash-output-elasticsearch-11.15.9-java/lib/logstash/outputs/elasticsearch/http_client/pool.rb:213] elasticsearch - Failed to perform request {:message=>"Certificate for <docker-es01-1> doesn't match any of the subject alternative names: [localhost, 127.0.0.1, es01]", :exception=>Manticore::UnknownException, :cause=>#<Java::JavaxNetSsl::SSLPeerUnverifiedException: Certificate for <docker-es01-1> doesn't match any of the subject alternative names: [localhost, 127.0.0.1, es01]>}

[WARN ] 2023-08-14 16:31:04.725 [Ruby-0-Thread-19: /usr/share/logstash/vendor/bundle/jruby/2.6.0/gems/logstash-output-elasticsearch-11.15.9-java/lib/logstash/outputs/elasticsearch/http_client/pool.rb:213] elasticsearch - Attempted to resurrect connection to dead ES instance, but got an error {:url=>"https://elastic:xxxxxx@docker-es01-1:9200/", :exception=>LogStash::Outputs::ElasticSearch::HttpClient::Pool::HostUnreachableError, :message=>"Elasticsearch Unreachable: [https://docker-es01-1:9200/][Manticore::UnknownException] Certificate for <docker-es01-1> doesn't match any of the subject alternative names: [localhost, 127.0.0.1, es01]"}

[WARN ] 2023-08-14 16:31:09.127 [Ruby-0-Thread-19: /usr/share/logstash/vendor/bundle/jruby/2.6.0/gems/logstash-output-elasticsearch-11.15.9-java/lib/logstash/outputs/elasticsearch/http_client/pool.rb:213] elasticsearch - Failed to perform request {:message=>"Certificate for <docker-es01-1> doesn't match any of the subject alternative names: [localhost, 127.0.0.1, es01]", :exception=>Manticore::UnknownException, :cause=>#<Java::JavaxNetSsl::SSLPeerUnverifiedException: Certificate for <docker-es01-1> doesn't match any of the subject alternative names: [localhost, 127.0.0.1, es01]>}

[WARN ] 2023-08-14 16:31:09.145 [Ruby-0-Thread-19: /usr/share/logstash/vendor/bundle/jruby/2.6.0/gems/logstash-output-elasticsearch-11.15.9-java/lib/logstash/outputs/elasticsearch/http_client/pool.rb:213] elasticsearch - Attempted to resurrect connection to dead ES instance, but got an error {:url=>"https://elastic:xxxxxx@docker-es01-1:9200/", :exception=>LogStash::Outputs::ElasticSearch::HttpClient::Pool::HostUnreachableError, :message=>"Elasticsearch Unreachable: [https://docker-es01-1:9200/][Manticore::UnknownException] Certificate for <docker-es01-1> doesn't match any of the subject alternative names: [localhost, 127.0.0.1, es01]"}

[WARN ] 2023-08-14 16:31:14.289 [Ruby-0-Thread-19: /usr/share/logstash/vendor/bundle/jruby/2.6.0/gems/logstash-output-elasticsearch-11.15.9-java/lib/logstash/outputs/elasticsearch/http_client/pool.rb:213] elasticsearch - Failed to perform request {:message=>"Certificate for <docker-es01-1> doesn't match any of the subject alternative names: [localhost, 127.0.0.1, es01]", :exception=>Manticore::UnknownException, :cause=>#<Java::JavaxNetSsl::SSLPeerUnverifiedException: Certificate for <docker-es01-1> doesn't match any of the subject alternative names: [localhost, 127.0.0.1, es01]>}

[WARN ] 2023-08-14 16:31:14.294 [Ruby-0-Thread-19: /usr/share/logstash/vendor/bundle/jruby/2.6.0/gems/logstash-output-elasticsearch-11.15.9-java/lib/logstash/outputs/elasticsearch/http_client/pool.rb:213] elasticsearch - Attempted to resurrect connection to dead ES instance, but got an error {:url=>"https://elastic:xxxxxx@docker-es01-1:9200/", :exception=>LogStash::Outputs::ElasticSearch::HttpClient::Pool::HostUnreachableError, :message=>"Elasticsearch Unreachable: [https://docker-es01-1:9200/][Manticore::UnknownException] Certificate for <docker-es01-1> doesn't match any of the subject alternative names: [localhost, 127.0.0.1, es01]"}

      
```

Thank you

**Comment by killmasta93** [comment]
hi @a3ilson I was wondering if you were able to replicate the same issue?

Thank you

**Comment by a3ilson** [comment]
@killmasta93 
My apologies, I thought you were all good…. But after looking over, I realized an oversight on my part. You need to amend the output as

hosts => ["https://es01:9200"]

**Comment by a3ilson** [comment]
Updated the 50-outputs.pfelk and will update the instructions in a few days (traveling).

**Comment by killmasta93** [comment]
hi @a3ilson no worries let me know so i can try it out again to leave the tutorial good for other users also

**Comment by a3ilson** [comment]
@killmasta93 

change/update the 50-outputs.pfelk with the following as your only “hosts” line

hosts => ["https://es01:9200"]

**Comment by killmasta93** [comment]
awesome its working
